deepspeed/launcher/runner.py

```diff
@@ -390,7 +390,7 @@ def main(args=None):
     args = parse_args(args)
 
     # For when argparse interprets remaining args as a single string
-    args.user_args = shlex.split(" ".join(args.user_args))
+    args.user_args = shlex.split(" ".join(list(map(lambda x: x if x.startswith("-") else f"'{x}'", args.user_args))))
```

Comment:
@mrwyattii - any reason why in #4007 we didn't copy the multinode_runner logic [here](https://github.com/microsoft/DeepSpeed/blob/0a0819b78554a5130b7c1b31d18779d39421dade/deepspeed/launcher/multinode_runner.py#L64)?

Reply:
Nope, but this change looks to be the proper fix